torchao/csrc/cuda/fp6_llm/fp6_linear.cu

```diff
@@ -14,6 +14,8 @@
 // 
 // This file is adapted from https://github.com/usyd-fsalab/fp6_llm/blob/5df6737cca32f604e957e3f63f03ccc2e4d1df0d/fp6_llm/csrc/fp6_linear.cu
 
+#if !defined(__CUDA_ARCH__) || __CUDA_ARCH__ >= 800  // at least Ampere
```

Comment:
For anyone wondering why this instead of `defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 800`, I'm following  [this](https://github.com/pytorch/pytorch/blob/e98eac76b358fb4639b9e9ce6894014354d7b073/aten/src/ATen/native/cuda/int4mm.cu#L1).

One reason I can think of is for vscode intellisense to work. If I use `#if defined(__CUDA_ARCH__)`, vscode will not do syntax highlighting. There are workarounds, like [this](https://forums.developer.nvidia.com/t/the-default-value-of-cuda-arch-macro-causes-the-vscode-intellisence-to-report-errors/300545/2), but it is more cumbersome.